cpp/inc/bond/core/config.h

```diff
@@ -5,63 +5,10 @@
 
 #include <boost/config.hpp>
 
-#if defined(BOOST_NO_CXX11_DEFAULTED_FUNCTIONS)
-#define BOND_NO_CXX11_DEFAULTED_FUNCTIONS
-#else
-  #if defined(_MSC_VER) && (_MSC_VER < 1900)
-  #define BOND_NO_CXX11_DEFAULTED_FUNCTIONS
-  #endif
-#endif
-
-#if defined(BOOST_NO_CXX11_AUTO_DECLARATIONS) || defined(BOOST_NO_AUTO_DECLARATIONS)
-#define BOND_NO_CXX11_AUTO_DECLARATIONS
-#endif
-
-#if defined(BOOST_NO_CXX11_AUTO_MULTIDECLARATIONS) || defined(BOOST_NO_AUTO_MULTIDECLARATIONS)
-#define BOND_NO_CXX11_AUTO_MULTIDECLARATIONS
-#endif
-
-#if defined(BOOST_NO_CXX11_LAMBDAS) || defined(BOOST_NO_LAMBDAS)
-#define BOND_NO_CXX11_LAMBDAS
-#endif
-
 #if defined(BOOST_NO_CXX11_NOEXCEPT) || defined(BOOST_NO_NOEXCEPT)
 #define BOND_NO_CXX11_NOEXCEPT
 #endif
 
-#if defined(BOOST_NO_CXX11_NULLPTR) || defined(BOOST_NO_NULLPTR)
-#define BOND_NO_CXX11_NULLPTR
-#endif
-
-#if defined(BOOST_NO_CXX11_RVALUE_REFERENCES) || defined(BOOST_NO_RVALUE_REFERENCES)
-#define BOND_NO_CXX11_RVALUE_REFERENCES
-#endif
-
-#if defined(BOOST_NO_CXX11_DEFAULTED_FUNCTIONS) \
-    || defined(BOOST_NO_CXX11_RVALUE_REFERENCES) \
-    || defined(BOOST_NO_RVALUE_REFERENCES)
-// We need support for both defaulted functions and rvalue references to use
-// default move ctors
-#define BOND_NO_CXX11_DEFAULTED_MOVE_CTOR
-#else
-    #if defined(_MSC_VER) && (_MSC_VER < 1900)
-    // Versions of MSVC prior to 1900 do not support = default for move ctors
-    #define BOND_NO_CXX11_DEFAULTED_MOVE_CTOR
```

Comment:
The code may be cleaner if we leave BOND_NO_CXX11_DEFAULTED_MOVE_CTOR in config.h instead of inlining it the generated code. Try it and see. I'll defer to your judgment here.

Reply:
I agree that it will be cleaner. I hope that going forward we will have less such checks and not more (this change removes most). Also, one argument against that is that it makes `gbc` more coupled with core library code. I'll probable choose to go this way given that there is only one such check left in `gbc`.

Reply:
Looks like `ifndef` can also be removed.